docs/csharp/programming-guide/concepts/linq/element-operations.md

```diff
@@ -19,8 +19,8 @@ Element operations return a single, specific element from a sequence.
 |FirstOrDefault|Returns the first element of a collection, or the first element that satisfies a condition. Returns a default value if no such element exists.|Not applicable.|<xref:System.Linq.Enumerable.FirstOrDefault%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.FirstOrDefault%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.FirstOrDefault%60%601%28System.Linq.IQueryable%7B%60%600%7D%29?displayProperty=nameWithType>|  
 |Last|Returns the last element of a collection, or the last element that satisfies a condition.|Not applicable.|<xref:System.Linq.Enumerable.Last%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Last%2A?displayProperty=nameWithType>|  
 |LastOrDefault|Returns the last element of a collection, or the last element that satisfies a condition. Returns a default value if no such element exists.|Not applicable.|<xref:System.Linq.Enumerable.LastOrDefault%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.LastOrDefault%2A?displayProperty=nameWithType>|  
-|Single|Returns the only element of a collection, or the only element that satisfies a condition.|Not applicable.|<xref:System.Linq.Enumerable.Single%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Single%2A?displayProperty=nameWithType>|  
-|SingleOrDefault|Returns the only element of a collection, or the only element that satisfies a condition. Returns a default value if no such element exists or the collection does not contain exactly one element.|Not applicable.|<xref:System.Linq.Enumerable.SingleOrDefault%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.SingleOrDefault%2A?displayProperty=nameWithType>|  
+|Single|Returns the only element of a collection or the only element that satisfies a condition. Throws an <xref:System.InvalidOperationException> if a collection is either empty or contains more than one element, or if there is no element or more than one element that satisfies a condition. |Not applicable.|<xref:System.Linq.Enumerable.Single%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.Single%2A?displayProperty=nameWithType>|  
+|SingleOrDefault|Returns the only element of a collection or the only element that satisfies a condition. Returns a default value if a collection is empty or no such element exists. Throws an <xref:System.InvalidOperationException> if a collection contains more than one element or more than one element satisfies a condition.|Not applicable.|<xref:System.Linq.Enumerable.SingleOrDefault%2A?displayProperty=nameWithType><br /><br /> <xref:System.Linq.Queryable.SingleOrDefault%2A?displayProperty=nameWithType>|  
```

Comment:
Would it make sense to simplify these so that they don't have to repeatedly talk about "collection or condition"? Maybe like the following?

Single:

> Throws an \<xref:System.InvalidOperationException> if there is no element or more than one element to return.

SingleOrDefault:

> Returns a default value if there is no element to return. Throws an \<xref:System.InvalidOperationException> if there is more than one element to return.

Reply:
@svick that is shorter and better. Thanks!

Reply:
Though "more than one element to return" may sound strange, I think it delivers the point.